--- a/lib/src/main/java/com/lody/virtual/client/hook/proxies/am/MethodProxies.java
+++ b/lib/src/main/java/com/lody/virtual/client/hook/proxies/am/MethodProxies.java
@@ -818,6 +818,7 @@
             } else {
                 VLog.e(getClass().getSimpleName(), "Unknown flag : " + args[4]);
             }
+            if (notification != null) notification = com.lody.virtual.client.hook.proxies.notification.NotificationChannelHooks.notification(notification);
             VNotificationManager.get().dealNotification(id, notification, getAppPkg());
 
             /**
@@ -1563,6 +1564,7 @@
                     IInterface provider = ContentProviderHolderOreo.provider.get(holder);
                     if (provider != null) {
                         provider = VActivityManager.get().acquireProviderClient(userId, info);
+                        if (provider != null) provider = ProviderHook.createProxy(false, info.authority, provider);
                     }
                     ContentProviderHolderOreo.provider.set(holder, provider);
                     ContentProviderHolderOreo.info.set(holder, info);
@@ -1570,6 +1572,7 @@
                     IInterface provider = IActivityManager.ContentProviderHolder.provider.get(holder);
                     if (provider != null) {
                         provider = VActivityManager.get().acquireProviderClient(userId, info);
+                        if (provider != null) provider = ProviderHook.createProxy(false, info.authority, provider);
                     }
                     IActivityManager.ContentProviderHolder.provider.set(holder, provider);
                     IActivityManager.ContentProviderHolder.info.set(holder, info);
