--- a/lib/src/main/java/com/lody/virtual/client/core/InvocationStubManager.java
+++ b/lib/src/main/java/com/lody/virtual/client/core/InvocationStubManager.java
@@ -46,6 +46,7 @@
 import com.lody.virtual.client.hook.proxies.pm.PackageManagerStub;
 import com.lody.virtual.client.hook.proxies.power.PowerManagerStub;
 import com.lody.virtual.client.hook.proxies.restriction.RestrictionStub;
+import com.lody.virtual.client.hook.proxies.role.RoleManagerStub;
 import com.lody.virtual.client.hook.proxies.search.SearchManagerStub;
 import com.lody.virtual.client.hook.proxies.shortcut.ShortcutServiceStub;
 import com.lody.virtual.client.hook.proxies.telephony.TelephonyRegistryStub;
@@ -199,6 +200,7 @@
             if (BuildCompat.isQ()) {
             	addInjector(new ActivityTaskManagerStub());
             	addInjector(new UriGrantsManagerStub());
+            	addInjector(new RoleManagerStub());
 
             	// http://aospxref.com/android-10.0.0_r47/xref/frameworks/base/core/java/android/os/IDeviceIdentifiersPolicyService.aidl#24
             	addInjector(new DeviceIdentifiersPolicyServiceStub());
